### PR TITLE
Non Blocking Accel Calibration for Gimbal and Copter

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 1]
```diff
@@ -454,6 +463,51 @@ bool AP_InertialSensor::_calculate_trim(const Vector3f &accel_sample, float& tri
...
+
+    Vector3f off;
+    Vector3f sf;
+    
+    _accel_cal[0].get_sample(0,float_imu);
+    _accel_cal[BODY_FIXED_IMU].get_sample(0,body_fixed);
+    //apply scale factors and offsets
+    off = _accel_offset[BODY_FIXED_IMU].get();
+    sf = _accel_scale[BODY_FIXED_IMU].get();
+    body_fixed += off;
```
Comment (jschall) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp]: needs to be a subtract

**Comment by cgkevork** [comment]
@bugobliterator Was this merged? The accel cal does not currently work on a Solo with a gimbal

**Comment by nick3dr** [comment]
It's a problem with iOS I think

On Thursday, August 27, 2015, Chris Kevorkian <notifications@github.com>
wrote:

> @bugobliterator <https://github.com/bugobliterator> Was this merged? The
> accel cal does not currently work on a Solo with a gimbal
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/3drobotics/ardupilot-solo/pull/133#issuecomment-135178658>
> .
>


-- 
*Nick Speal*
Engineering Project Manager
*3D Robotics*

1608 4th Street, Suite 410
Berkeley, CA 94710
website <http://www.3dr.com/> | facebook
<http://www.facebook.com/3drobotics> | instagram
<http://www.instagram.com/3drobotics>

**Comment by cgkevork** [comment]
It is, apparently Android is fine. IOS just needs a minor tweek.

**Pull request closed without merging**